### PR TITLE
[aotinductor] Allow specifying a .so name in the aot_inductor.output_path config

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112728
* __->__ #112651

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

Differential Revision: [D50902585](https://our.internmc.facebook.com/intern/diff/D50902585)

### CODE REVIEW

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D50902585).

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D50902585).

File: torch/_inductor/codecache.py [hunk 1]
```diff
@@ -1545,14 +1556,24 @@ def compile(
...
-            specified_dir=config.aot_inductor.output_path,
+            specified_dir=specified_output_path,
         )
 
-        if key not in cls.cache:
+        if key not in cls.cache or (
+            specified_output_path
+            and os.path.dirname(cls.cache[key]) != specified_output_path
+            or specified_so_name
+            and os.path.basename(cls.cache[key]) != specified_so_name
```
Comment (chenyang78) [inline torch/_inductor/codecache.py:1575]: We cache the full path for the output.so, which is either ```config.aot_inductor.output_path``` or ```os.path.splitext(input_path)[0] + ".so"```. So, seems we wouldn't have any case where we cache ```specified_so_name```?

**Review by chenyang78 — APPROVED** [review]
Left a minor comment. LGTM overall. Thanks!

**Pull request closed without merging**